src/HalResource.php

```diff
@@ -30,6 +30,8 @@
  * The class name "HalResource" was chosen as "resource" has been given a
  * soft designation as a future keyword in PHP as of PHP 7; choosing this
  * name now makes the class future-proof.
+ *
+ * @final
```

Comment:
Unsure about resources being `final`, since it's not interfaced, and not really a VO?